release.js

```diff
@@ -0,0 +1,121 @@
+import chalk from 'chalk';
+import esMain from 'es-main';
+import fs from 'fs-extra';
+import inquirer from 'inquirer';
+
+import {exec} from './scripts.js';
+
+const currentVersion = (
+  await fs.readJson(new URL('./package.json', import.meta.url))
+).version;
+
+const getNewVersion = async () => {
+  const versionParts = currentVersion.split('.').map((x) => Number(x));
+  const newVersions = [
+    `${versionParts[0] + 1}.0.0`,
+    `${versionParts[0]}.${versionParts[1] + 1}.0`,
+    `${versionParts[0]}.${versionParts[1]}.${versionParts[2] + 1}`
+  ];
+
+  const answers = await inquirer.prompt([
+    {
+      type: 'list',
+      name: 'newVersion',
+      message: 'How should we bump the version?',
+      choices: [
+        {
+          name: `Major (${newVersions[0]})`,
+          value: newVersions[0]
+        },
+        {
+          name: `Minor (${newVersions[1]})`,
+          value: newVersions[1]
+        },
+        {
+          name: `Patch (${newVersions[2]})`,
+          value: newVersions[2]
+        }
+      ],
+      default: 2
+    }
+  ]);
+
+  return answers.newVersion;
+};
+
+const doVersionBump = async (newVersion) => {
+  for (const f of ['./package.json', './package-lock.json']) {
+    const filepath = new URL(f, import.meta.url);
+    const data = await fs.readJson(filepath);
+    data.version = newVersion;
+    await fs.writeJson(filepath, data, {spaces: 2});
+  }
+};
+
+const getChanges = () => {
+  const changes = exec(`git log --pretty=reference v${currentVersion}..main`)
+    .toString('utf8')
+    .split('\n')
+    .map((c) => '- ' + c.substring(8));
+  changes.pop();
+  return changes.join('\n');
+};
+
+const doChangelogUpdate = async () => {
+  const filepath = new URL('./CHANGELOG.md', import.meta.url);
+  const changelog = await fs.readFile(filepath, 'utf8');
+  const idx = changelog.indexOf('##');
+  const newChangelog =
+    changelog.substring(0, idx) +
+    getChanges() +
+    '\n\n' +
+    changelog.substring(idx, changelog.length);
+  await fs.writeFile(filepath, newChangelog, 'utf8');
+};
+
+const prepareBranch = (newVersion) => {
+  const branch = `release-${newVersion}`;
+
+  exec('git checkout main');
```

Comment:
This step fails for me, the script just made local changes and checkout fails with "Your local changes to the following files would be overwritten by checkout".

I think the script should first ensure that there are no local changes to begin with (before making the bump changes), then check out the new release branch, and only after that list changes to make the changelog.